Assets/Plugins/Source/Core/RuntimeConfig.cs

```diff
@@ -0,0 +1,332 @@
+            InitializeThreadAffinity threadAffinity = new();
+            config.ConfigureOverrideThreadAffinity(ref threadAffinity);
+
+            // try parse the float values from string so that they default to
+            // zero if the strings are empty.
+            _ = float.TryParse(config.initialButtonDelayForOverlay, out float initialButtonDelayForOverlayFloat);
+            _ = float.TryParse(config.repeatButtonDelayForOverlay, out float repeatButtonDelayForOverlayFloat);
+
+            return new RuntimeConfig(
+                Guid.Parse(config.productID),
+                config.productName,
+                Version.Parse(config.productVersion),
+                config.sandboxID,
+                Guid.Parse(config.deploymentID),
+                new ClientCredentials() { ClientId = config.clientID, ClientSecret = config.clientSecret },
+                config.encryptionKey,
+                config.GetPlatformFlags(),
+                config.GetAuthScopeFlags(),
+                config.GetIntegratedPlatformManagementFlags(),
+                config.tickBudgetInMilliseconds,
+                threadAffinity,
+                config.isServer,
+                config.alwaysSendInputToOverlay,
+                initialButtonDelayForOverlayFloat,
+                repeatButtonDelayForOverlayFloat,
+                config.hackForceSendInputDirectlyToSDK);
+        }
+        
+        public override bool Equals(object obj)
+        {
+            if (obj is RuntimeConfig other)
+            {
+                return ProductName == other.ProductName &&
+                       ProductVersion.Equals(other.ProductVersion) &&
+                       ProductId.Equals(other.ProductId) &&
+                       SandboxId == other.SandboxId &&
+                       DeploymentId.Equals(other.DeploymentId) &&
+                       ClientCredentials.Equals(other.ClientCredentials) &&
+                       EncryptionKey == other.EncryptionKey &&
+                       PlatformFlags == other.PlatformFlags &&
+                       AuthScopeFlags == other.AuthScopeFlags &&
+                       IntegratedPlatformManagementFlags == other.IntegratedPlatformManagementFlags &&
+                       TickBudgetInMilliseconds == other.TickBudgetInMilliseconds &&
+                       ThreadAffinity.Equals(other.ThreadAffinity) &&
+                       IsServer == other.IsServer &&
+                       AlwaysSendInputToOverlay == other.AlwaysSendInputToOverlay &&
+                       Math.Abs(InitialButtonDelayForOverlay - other.InitialButtonDelayForOverlay) < FloatEqualityTolerance &&
+                       Math.Abs(RepeatButtonDelayForOverlay - other.RepeatButtonDelayForOverlay) < FloatEqualityTolerance &&
+                       SendInputDirectlyToSDK == other.SendInputDirectlyToSDK;
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            HashCode hash = new();
```

Comment:
oh WHAT. I'm using this from now on when making HashCodes, I was super tired of making up baloney on the spot. Had no idea you could just do this!